src/duct/core/repl.clj

```diff
@@ -17,3 +19,31 @@
   fipp/IOverride
   fipp/IEdn
   (-edn [r] (tagged-literal 'duct/resource (:path r))))
+
+(def ^:private event->file-name
+  (comp #(.getName %)
+        :file))
+
+(def ^:private matches-clojure-file?
+  (partial re-matches #"^[a-zA-Z]{1}[a-zA-Z_]*(.clj|.edn)$"))
+
+(defn- clojure-file?
+  [ctx event]
+  (-> event
+      event->file-name
+      matches-clojure-file?))
+
+(defn- auto-reset-handler
+  [ctx event]
+  (binding [*ns* true]
+    (in-ns 'integrant.repl)
+    (integrant.repl/reset))
+  ctx)
+
+(defn auto-reset
+  "Automatically reset Duct configuration on file change.
```

Comment:
I think this would be better worded as:

```clojure
  "Automatically reset the system when a Clojure or edn file is changed in
  `src` or `resources`."
```

This should also handle `dev/src` and `dev/resources` as well.